src/poly/poly.jl

```diff
@@ -711,7 +718,7 @@ end
 
 function (x::SPolyUnion{T} == y::SPolyUnion{T}) where T <: Nemo.RingElem
    check_parent(x, y)
-   GC.@preserve x y return Bool(libSingular.p_EqualPolys(x.ptr, y.ptr, parent(x).ptr))
+   return(iszero(x-y))
 end
 
 function Base.isless(x::SPolyUnion{T}, y::SPolyUnion{T}) where T <: Nemo.RingElem
```

Comment:
Do we also need something similar here?

Reply:
I don't think so: the classes of 0 and 1 are special (and therefore also the operator ==), but otherwise
we work with any representat of the class (in the quotinet ring case)